boj/silver/지름길/minjong.js

```diff
@@ -0,0 +1,98 @@
+const isLinux = process.platform === 'linux'
+const file = isLinux ? '/dev/stdin' : '../../input.txt';
+const input = require('fs')
+  .readFileSync(file, 'utf-8')
+  .trim()
+  .split('\n');
+
+
+const solution = (input) => {
+  const [N, D] = input[0].split(' ').map(Number);
+  
+  const graph = Array.from({ length: D + 1}, () => []);
+  for (let i = 1; i < input.length; i += 1){
+    const [a, b, c] = input[i].split(' ').map(Number);
+    if (D < b) continue;
+    if (b - a <= c) continue;
+    graph[a].push([b, c]);
+  }
+
+  const distance = Array.from({length: D + 1}, () => Infinity);
+  const heap = new MinHeap();
+
+  distance[0] = 0;
+
+  for (let i = 0; i <= D; i += 1) {
+    if (i > 0) distance[i] = Math.min(distance[i], distance[i - 1] + 1);
+
+    heap.push([i, distance[i]]);
+    while (!heap.isEmpty()) {
+      const [src, _] = heap.pop();
+
+      for (const [dst, nextDistance] of graph[src]) {
+        if (distance[dst] > distance[src] + nextDistance) {
+          distance[dst] = distance[src] + nextDistance;
+          heap.push([dst, distance[dst]]);
```

Comment:
`heap`에 저장한 배열의 두 번째 요소는 사용하지 않기 때문에, 제거해도 되지 않을까 싶습니다! 🙄
```suggestion
          const src = heap.pop();
          ...
          heap.push(dst);
```

Reply:
push, pop할 때 우선순위를 판별하기 위해 사용하고 있습니다!